### PR TITLE
Server: Update API routing

### LINKED ISSUE #14: Server: Update API routing
# Description

Let's stay on the `Bun` run and use [`ElysiaJS`](https://elysiajs.com/) for our server. It has some nice error handling and should be lightning fast.

Also, it's about time we start dynamically returning content in the user's language rather than just returning the default translation.

# Acceptance criteria

- [x] Swap out the existing `Express.js` server for [`ElysiaJS`](https://www.npmjs.com/package/elysia)
- [x] Use the request headers to determine which language to return
- [x] Sit the API behind a `/api` route

### PR DESCRIPTION
**Pull request by bryceknz**
- [➕ Swap `Express.js` for `ElysiaJS`](https://github.com/bryceknz/sanity-translations-experiment/commit/a4f7171b4b08edcd257398cd4e9cfef112517873)
- [♻️ Re-write server routing with `ElysiaJS`](https://github.com/bryceknz/sanity-translations-experiment/commit/f429ade598ec952c7ef156a027199ce85ce1600e)
- [✨ Translate content dynamically](https://github.com/bryceknz/sanity-translations-experiment/commit/cfc7b4897bc61966c5a4417b4b773bbc30d7f0f3)
- [➕ Added `@elysiajs/cors`](https://github.com/bryceknz/sanity-translations-experiment/pull/15/commits/0e1ffdb6547e8e3c4a55430e5635e593e5898879)
- [🔒 Enabled CORS for `localhost`](https://github.com/bryceknz/sanity-translations-experiment/pull/15/commits/6a4081759d0c9472d67c66a9e6b1984128611c9a)

Resolves #14.

### CODE REVIEW

File: server/index.ts
Comment (umxr) [inline server/index.ts:1]: Could we add in the CORS plugin? https://elysiajs.com/plugins/cors.html - i had to update the express server to allow the FE to connect to the server. I presume it should just be a case of following the setup and 🤞🏽 it should just work

**Review by umxr — CHANGES_REQUESTED** [review]
just on tweak, otherwise its all good 👍🏽. ElysiaJS is pretty neat

File: server/index.ts
Comment (bryceknz) [reply server/index.ts:1]: Ooh great idea! Does this look alright to you? My hope is it should allow any requests from `http://localhost:*` 🤞🏻:
```ts
const app = new Elysia()
  .use(cors({ origin: /^http:\/\/localhost:\d+$/ }))
```

Comment (umxr) [reply server/index.ts:1]: Looks like it should work 🤞

Comment (bryceknz) [reply server/index.ts:1]: Fixed: [🔒 Enabled CORS for `localhost`](https://github.com/bryceknz/sanity-translations-experiment/pull/15/commits/6a4081759d0c9472d67c66a9e6b1984128611c9a)

**Review by umxr — APPROVED** [review]
LGTM

**Pull request merged**